### PR TITLE
Use code to generate Unicode-LaTeX character mapping table

### LINKED ISSUE #218: Replace Unicode / LaTeX conversion table with R script
Suggestion from @yihui: https://github.com/Merck/r2rtf/pull/217/files?diff=unified&w=0#r1564779269

The goal is to put the `unicode_latex` variable in an R script, to replace the current binary form stored in `R/sysdata.rda`.

This would also avoid the need of the `globalVariables()` call for `unicode_latex` as a hack to bypass R CMD check notes.

### PR DESCRIPTION
**Pull request by nanxstats**
Fixes #218 

This PR creates an internal function in `R/utils.R` to generate the mapping table into `R/unicode_latex.R`.

This eliminates the need for using the binary file `sysdata.rda` and is more friendly for version control.

The new, code-generated data frame is bitwise identical to the version saved in `sysdata.rda`, except that the `int` column is of class integer, not numeric.

## Data ingestion issue worth following up

You might want to check the data ingestion logic. I found no evidence on how the previous version was constructed. I used some _ad hoc_ logic to get an identical version of the table, but it would be good to check if the data included in the previous version is reasonable, or what specific filters were applied. For example, from the beginning, without using `quote = ""` in `read.table()`, it will give:

```
Warning message:
In scan(file = file, what = what, sep = sep, quote = quote, dec = dec,  :
  EOF within quoted string
```

This will result in only 1740 rows vs. 2757 rows when using `quote = ""`, which avoids the warning.

### CODE REVIEW

**Comment by nanxstats** [comment]
@yihui in case you got a minute to review

File: R/utils.R [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    seq_len(nrow(tbl)),
+    function(i) {
+      sprintf(
+        '  c(unicode = "%s", latex = "%s", int = %d)',
+        tbl$unicode[i],
+        gsub("\\", "\\\\", tbl$latex[i], fixed = TRUE),
+        tbl$int[i]
+      )
+    }
+  )
```
Comment (yihui) [inline R/utils.R:67]:
```suggestion
  rows <- paste(
    tbl$unicode,
    gsub("\\", "\\\\", tbl$latex, fixed = TRUE),
    sprintf("%d", tbl$int),
    sep = ", "
  )
```

File: R/utils.R [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  writeLines(c(
+    "# Generated by R/utils.R: do not edit by hand",
+    "# Please run update_unicode_latex() to regenerate this file",
+    "",
+    "unicode_latex <- data.frame(rbind(",
+    paste(rows, collapse = ",\n"),
+    "), stringsAsFactors = FALSE)",
```
Comment (yihui) [inline R/utils.R:75]: I didn't really test my code.

```suggestion
    "unicode_latex <- matrix(c(",
    paste0("  ", rows, collapse = ",\n")
    "), ncol = 3, byrow = TRUE)",
    "unicode_latex <- as.data.frame(unicode_latex)",
    'names(unicode_latex) <- c("unicode", "latex", "int")',
```

**Review by yihui — APPROVED** [review]
First, I'd prefer using a `matrix` to write the data, which is a little more compact than the data frame.

Second, I wonder if it's worth the effort to make the file `R/unicode_latex.R` human-readable. If not, we could consider just `dump()` the data frame in `update_unicode_latex()`.

I don't have a strong opinion on either point. It's fine to merge the current PR as is.

**Comment by nanxstats** [comment]
> First, I'd prefer using a `matrix` to write the data, which is a little more compact than the data frame.
> 
> Second, I wonder if it's worth the effort to make the file `R/unicode_latex.R` human-readable. If not, we could consider just `dump()` the data frame in `update_unicode_latex()`.
> 
> I don't have a strong opinion on either point. It's fine to merge the current PR as is.

Great! Thanks. I've applied the changes and updated the table. The matrix version is exactly what we need to be less tedious. How I hoped there could be a row-wise data frame constructor in base. 😂 

Making it human-readable seems to be manageable in this case, so let's just keep it that way.

**Review by yihui — APPROVED** [review]

**Review by elong0527 — APPROVED** [review]
LGTM, thanks for improving the transparency the the source data!

File: R/utils.R [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  rows <- paste(
+    sprintf(
+      '"%s", "%s", %d',
+      tbl$unicode,
+      gsub("\\", "\\\\", tbl$latex, fixed = TRUE),
+      tbl$int
+    ),
+    sep = ", "
+  )
```
[the remark this answers is not in the document]
Comment (nanxstats) [reply R/utils.R:64]: Yes! Patched in another PR: https://github.com/Merck/r2rtf/pull/224

**Pull request merged**